Library/Homebrew/cmd/formulae.sh

```diff
@@ -16,9 +16,7 @@ homebrew-formulae() {
            -name vendor \
           \) \
          -prune -false -o -name '*\.rb' | \
-    sed -r -e 's/\.rb//g' \
-           -e 's_.*/Taps/(.*)/(home|linux)brew-_\1/_' \
-           -e 's|/Formula/|/|' \
+    sed 's:.*/Taps/\([^/]*\)/[hl][oi][mn][eu]x*brew-\([^/]*\).*/\(.*\).rb:\1/\2/\3:'
```

Comment:
This is dramatically harder to read. Could we pass `-r` or `-E` depending on the operating system instead?

Reply:
> Could we pass `-r` or `-E` depending on the operating system instead?

Yes we 🇨🇦  I'll update the PR.

Reply:
Here's a solution using `awk`:
```suggestion
    awk '{ \
      sub(".*/Taps/",""); \
      sub("/(home|linux)brew-","/"); \
      sub("/Formula/","/"); \
      sub("\\.rb",""); \
      print \
    }' \
```

Reply:
I'm OK with switching to `awk` but I think it might be better to stick with `sed` because it is an essential system component and `awk` is not (`awk` can be absent on some OSes).

Reply:
Seems like those OSes are quite rare or domain-specific, like BusyBox. Anything that claims to be UNIX or Linux [will have it](https://unix.stackexchange.com/q/82409).

Reply:
Would also rather avoid `awk` if possible.

Reply:
*sad `awk` noises*

Reply:
> Could we pass `-r` or `-E` depending on the operating system instead?

I updated the PR by moving `shims/mac/super/sed` shim to `shims/super/sed`, adding a section that handles `-E` flag, and using that shim in `brew formulae`. Does it seem reasonable?

Reply:
@maxim-belkin Please make the `sed` change inline here instead, I don't think it's necessary to add a shim when we've had no build process problems with this, thanks!

Reply:
OK!

Reply:
done!